### PR TITLE
Fix  reporting tools

### PR DESCRIPTION
**Pull request by qifeng-bai**
Incompatible query due to recent Postgres updates

### CODE REVIEW

File: grails-app/controllers/au/org/ala/spatial/service/ReportController.groovy [hunk 1]
```diff
@@ -25,18 +27,25 @@ class ReportController {
 
         //render taskService.report() as JSON
     }
-    def task(){
+
+    def task() {
+        boolean includeAll = false
+        if (params.containsKey("includeAll") && !params.includeAll) {
```
Comment (Rita-C) [inline grails-app/controllers/au/org/ala/spatial/service/ReportController.groovy:33]: Hard coded "includeAll"? Case sensitive?

**Review by Rita-C — CHANGES_REQUESTED** [review]

File: grails-app/services/au/org/ala/spatial/service/ReportService.groovy [hunk 2]
```diff
@@ -76,22 +76,23 @@ class ReportService {
...
-
-        String[] idsInALA = "1901,92092,71707,1493,53099,34,9965,47075,27078,44,9200,82292,10549,9048,11359,61,35022,13,28057,19807,27583,27190,35,48604,47326,35774,8443,8516,46293,4228,89299,89300,28712".split(',').collect{ '"' + it + '"'}
-        String idSql = idsInALA.join(',')
+    def taskBasedReport(includeAll){
+
+        String sql = "SELECT CONCAT(category1, '->', category2) as name, count(*) as count,extract(YEAR from created) AS year,  extract(MONTH from created) AS month  " +
+                        "FROM Log "
+        //Exclude ALA internal users
+        if (!includeAll) {
+            String[] idsInALA = "1901,92092,71707,1493,53099,34,9965,47075,27078,44,9200,82292,10549,9048,11359,61,35022,13,28057,19807,27583,27190,35,48604,47326,35774,8443,8516,46293,4228,89299,89300,28712".split(',').collect{ "'" + it + "'"}
```
[the remark this answers is not in the document]
Comment (qifeng-bai) [reply grails-app/services/au/org/ala/spatial/service/ReportService.groovy:85]: If we move our ALA IDs, we'll have lots meaningless data, e.g actions from integration tests, tests done by other developers etc

File: grails-app/controllers/au/org/ala/spatial/service/ReportController.groovy (hunk 1 above)
Comment (qifeng-bai) [reply grails-app/controllers/au/org/ala/spatial/service/ReportController.groovy:33]: Case sensitive on purpose.  The default and preferred way is to excludes some ALA members

File: grails-app/controllers/au/org/ala/spatial/service/ReportController.groovy [hunk 3]
```diff
@@ -25,18 +27,25 @@ class ReportController {
...
         try {
-            for(List report in reports)
-                outputStream << report.join(',') + "\r\n"
-        } catch (IOException e){
+            for (List report in reports) {
+                if (report[0])
+                    outputStream << report.join(',') + "\r\n"
+            }
+        } catch (IOException e) {
             null
```
[the remark this answers is not in the document]
Comment (qifeng-bai) [reply grails-app/controllers/au/org/ala/spatial/service/ReportController.groovy:46]: Http IO is unlikely to have this IOException, but anyway I added a log

File: grails-app/controllers/au/org/ala/spatial/service/ReportController.groovy (hunk 1 above)
Comment (Rita-C) [reply grails-app/controllers/au/org/ala/spatial/service/ReportController.groovy:33]: Ok

**Review by Rita-C — APPROVED** [review]

**Pull request merged**